### PR TITLE
Add Logback encoder

### PR DESCRIPTION
**Pull request by sabertiger**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -85,6 +85,16 @@
...
+            <groupId>ch.qos.logback</groupId>
+            <artifactId>logback-core</artifactId>
+            <version>1.1.11</version>
+        </dependency>
+        <dependency>
+            <groupId>ch.qos.logback</groupId>
+            <artifactId>logback-classic</artifactId>
+            <version>1.1.11</version>
+        </dependency>
+        <dependency>
```
Comment (sanjeevchopra) [inline pom.xml]: @FrankGasparovic will this cause any issues with slf4j  for apps which use log4j ?

**Comment by sanjeevchopra** [comment]
Please increment this artifact's minor version for this change.

File: pom.xml (hunk 1 above)
Comment (FrankGasparovic) [inline pom.xml]: We had issues when both log4j and logback were in the class path when we were working on dynamic logger. Although that issue (I think) stemmed from us changing both logging frameworks log level programmatically. @sabertiger did you test this version with your service?

Comment (rjtethered) [inline pom.xml]: @sanjeevchopra, we tested with logback. Anyone using log4j, may have to exclude logback dependencies explicitly. Hope, FrankGasparovic can do further integration test after Pull.

File: src/main/java/com/ge/predix/logback/PredixEncoder.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        logFormat.put("dpmt", mdc.getOrDefault("APP_ID", ""));
+        logFormat.put("inst", mdc.getOrDefault("INSTANCE_ID", ""));
+        logFormat.put("tid", event.getThreadName());
+        logFormat.put("mod", FILE_OF_CALLER_CONVERTER.convert(event));
+        if (null != event.getLevel()) {
+            logFormat.put("lvl", event.getLevel().toString());
+        }
+        logFormat.put("msg", event.getMessage());
+        if (null != event.getThrowableProxy()) {
+            logFormat.put("stck", getStackTrace(event));
```
Comment (sanjeevchopra) [inline src/main/java/com/ge/predix/logback/PredixEncoder.java]: 'stck' should be a JSON collection of strings, where each string is a individual line in the stack. Not a single string field.  Also please remove the "\n\t" from each line. That would be up to whatever is rendering the stack.

Comment (sabertiger) [inline src/main/java/com/ge/predix/logback/PredixEncoder.java]: @sanjeevchopra  done

**Pull request merged**